src/parser.cpp

```diff
@@ -376,54 +365,21 @@ namespace ada::parser {
    */
   std::optional<uint64_t> parse_ipv4_number(std::string_view input) {
     // If input is the empty string, then return failure.
-    if (input.empty()) {
-      return std::nullopt;
+    if (input.empty()) { return std::nullopt; }
+    // using std::strtoll is potentially dangerous if the string is not null terminated.
+    uint64_t result{};
```

Comment:
I used `uint64_t` due to an overflow issue, even though ipv4 numbers are uint32. I think we can revert it to 32 right now

Reply:
It is unlikely to matter much as far as performance goes. Sometimes, working on 64-bit integers is actually faster.